packages/utils/src/formatDataValueByType.js

```diff
@@ -3,19 +3,97 @@
  * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
  */
 
+import numeral from 'numeral';
 import { VALUE_TYPES } from './constant';
 
+/**
+ *
+ * @param {string|number} number number to truncate
+ * @param {string|number} decimalPlace how many decimal places to truncate number to
+ */
+const truncateDecimalToPlace = decimalPlace => number => {
+  const place = 10 ** decimalPlace;
+  return Math.floor(number * place) / place;
+};
+
+// Note: will display 0 if passed undefined
+const currency = value => numeral(value).format('$0.00a');
+
+const fraction = (value, { total }) => {
+  if (isNaN(total)) return 'No data';
```

Comment:
Since this method has now moved in the more generic `utils` package, I'd suggest that it also returns a more generic value (`null`/`undefined`). Then the consumer can decide on how to interpret it (display `No data` or other).

This is just an example of the general issue of how we handle `No data` values. I believe we currently do so in a bit of a hard coded way, which is not ideal. Since it's a general issue and may be tricky to handle correctly, I'm happy for this to stay as is for now - but please keep it in mind for a possible future refactor!

Reply:
Yes that's a good point. The no data string is a presentation concern and not a util concern. As you say I'm going to leave this one for now because it might be better to refactor the no data cases all at once or something.